docs/asciidoc/advanced-contextPropagation.adoc

```diff
@@ -5,20 +5,38 @@ Since 3.5.0, Reactor-Core embeds support for the `io.micrometer:context-propagat
 This library is intended as a means to easily adapt between various implementations of the concept of a Context, of which
 `ContextView`/`Context` is an example, and between `ThreadLocal` variables as well.
 
-`ReactorContextAccessor` allows the Context-Propagation library to understand Reactor `Context` and `Contextview`.
+`ReactorContextAccessor` allows the Context-Propagation library to understand Reactor
+`Context` and `ContextView`.
 It implements the SPI and is loaded via `java.util.ServiceLoader`.
 No user action is required, other than having a dependency on both reactor-core and `io.micrometer:context-propagation`. The `ReactorContextAccessor` class is public but shouldn't generally be accessed by user code.
 
-On top of that, Reactor-Core 3.5.0 also modifies the behavior of a couple key operators as well as introduces the `contextCapture` operator
-to transparently deal with `ContextSnapshot`s if the library is available at runtime.
+Reactor-Core supports two modes of operation with `io.micrometer:context-propagation`:
 
-== `contextCapture` Operator
+- the **default** (limited) mode,
+- and the **automatic** mode, enabled via `Hooks.enableAutomaticContextPropagation()`.
+  Please note that this mode applies only to new subscriptions, so it is recommended to
+enable this hook when the application starts.
+
+Their key differences are discussed in the context of either <<context-writing, writing
+data>> to Reactor `Context`, or <<context-accessing,accessing `ThreadLocal` state>> that
+reflects the contents of the`Context` of currently attached `Subscriber` for reading.
+
+The act of discovering the mapping between `ThreadLocal` values and relevant entries in
+the Reactor `Context` is performed using the registered `ThreadLocalAccessor`(s).
+
+[[context-writing]]
+== Writing to `Context`
+
+Depending on the individual application, you might either have to store already populated
+`ThreadLocal` state as entries in the `Context`, or might only need to directly populate
+the `Context`.
+
+=== `contextCapture` Operator
 This operator can be used when one needs to capture `ThreadLocal` value(s) at subscription time and reflect these values in the Reactor `Context` for the benefit of upstream operators.
-It relies on the `context-propagation` library and notably the registered `ThreadLocalAccessor`(s) to discover relevant ThreadLocal values.
 
 This is a convenient alternative to `contextWrite` which uses the `context-propagation` API to obtain a `ContextSnapshot` and then uses that snapshot to populate the Reactor `Context`.
```

Comment:
btw. Maybe it is me but I reread multiple times until I got that it is about contextCapture and not contextWrite which uses contextRegistry. may be worth rephrasing as well to something like: 

"In opposite to manual `contextWrite` `contextCapture` uses the `context-propagation` API ...`